test/realm_object_test.dart

```diff
@@ -357,4 +362,17 @@ Future<void> main([List<String>? args]) async {
     // RemappedClass is mapped as `myRemappedClass`
     expect(json, contains('"property with spaces":{ "table": "class_myRemappedClass", "key": 0}'));
   });
+
+  test('RealmObject read/write bool value', () {
+    var config = Configuration([BoolValue.schema]);
+    var realm = getRealm(config);
+
+    realm.write(() {
+      realm.add(BoolValue(true));
+      realm.add(BoolValue(false));
+    });
+
+    expect(realm.all<BoolValue>()[0].value, true);
+    expect(realm.all<BoolValue>()[1].value, false);
```

Comment:
Realm does not guarantee ordering of tables. In most cases, the order will be the insertion order, but we can't depend on this behavior not changing.

Reply:
ok I will redo using primary keys. 
also I have verified the fix manually.